homework/src/index.js

```diff
@@ -30,18 +32,113 @@
+    span1.innerText = 'Repository: ';
+    const p1 = createAndAppend('p', li);
+    const a = createAndAppend('a', p1, { href: info.RepositoryUrl, target: '_blank' });
+    a.innerText = `${info.Repository}`;
+    const li2 = createAndAppend('li', ul);
+    const span2 = createAndAppend('span', li2, { id: 'title' });
+    span2.innerText = 'Description:  ';
+    const p2 = createAndAppend('p', li2)
+    p2.innerText = info.Description;
+    const li3 = createAndAppend('li', ul);
+    const span3 = createAndAppend('span', li3, { id: 'title' });
+    span3.innerText = 'Forks: ';
+    const p3 = createAndAppend('p', li3)
+    p3.innerText = info.Forks;
+    const li4 = createAndAppend('li', ul);
+    const span4 = createAndAppend('span', li4, { id: 'title' });
+    span4.innerText = 'Updated: ';
+    const p4 = createAndAppend('p', li4)
+    p4.innerText = info.Updated;
+  }
+
+  function renderContributorsInfo(info) {
+    fetchJSON(info.ContributorsUrl)
+      .then(data => {
+        ul2.innerText = '';
+        data.forEach(elem => {
+          const li = createAndAppend('li', ul2);
+          createAndAppend('img', li, { src: elem.avatar_url });
+          const a = createAndAppend('a', li, { href: elem.html_url, target: '_blank' });
+          a.innerText = `${elem.login}`;
+          const span = createAndAppend('span', li);
+          span.innerText = `${elem.contributions}`;
+        });
+      })
+      .catch(err => {
+        createAndAppend('div', div2, { text: err.message, class: 'alert-error' });
+      });
+  }
+
   function main(url) {
-    fetchJSON(url, (err, data) => {
-      const root = document.getElementById('root');
-      if (err) {
+    fetchJSON(url)
+      .then(data => {
+        const select = createAndAppend('select', header);
+        const notSortedQueries = [];
+        data.forEach(elem => {
+          const item = { Repository: elem.name, Description: elem.description, Forks: elem.forks, Updated: elem.updated_at, ContributorsUrl: elem.contributors_url, RepositoryUrl: elem.html_url };
+          notSortedQueries.push(item);
+        });
+        const queries = notSortedQueries.sort(function (a, b) {
+          const nameA = a.Repository.toUpperCase();
+          const nameB = b.Repository.toUpperCase();
+          if (nameA < nameB) {
+            return -1;
+          }
+          if (nameA > nameB) {
+            return 1;
+          }
+          return 0;
+        });
+        // console.log(notSortedQueries);
+        // console.log(queries);
+        queries.forEach((query, index) => {
+          createAndAppend('option', select, { value: index, text: query.Repository });
+        });
+
+        renderRepositoryInfo(queries[0]);
+
+        renderContributorsInfo(queries[0]);
+
+        select.addEventListener('change', () => {
+          // console.log(select.value);
+
+          // create the lift div1
+          const que = queries[select.value];
```

Comment:
The name `que` can be mistaken to mean `queue`, which is a completely different concept in computing from `query`. Avoid such confusing abbreviations. If you have an array of `queries` then a single element is called a `query`.